test/firtool/async-reset-anno.fir

```diff
@@ -0,0 +1,19 @@
+; RUN: firtool %s -mlir-print-ir-after=firrtl-infer-resets 2>&1 | FileCheck %s --check-prefix POST-INFER-RESETS
+; RUN: firtool %s -mlir-print-ir-after=firrtl-sfc-compat 2>&1 | FileCheck %s --implicit-check-not POST-SFC-COMPAT-NOT
```

Comment:
It's a bit janky to do this with `firtool -mlir-print-ir-after`. How about this:

```suggestion
; RUN: firtool %s -parse-only | circt-opt -firrtl-infer-resets | FileCheck %s --check-prefix POST-INFER-RESETS
; RUN: firtool %s -parse-only | circt-opt -firrtl-infer-resets -firrtl-sfc-compat | FileCheck %s --implicit-check-not POST-SFC-COMPAT-NOT
```

Reply:
As a further alternative, this could be two checks on the passes individually:

1. An MLIR test that infer resets doesn't delete the annotation. (The inversion of the test that was removed.)
2. A test that SFC Compat does remove the annotation.

I'm fine with either the more end-to-end approach here or the very directed MLIR tests in (1) and (2).

Reply:
i might keep these tests, i think the end-to-end approach is slicker